sdcm/utils/microbenchmarking/perf_simple_query_reporter.py

```diff
@@ -0,0 +1,149 @@
+import os
+import json
+from sdcm.results_analyze import BaseResultsAnalyzer, collections, TestConfig
+
+
+def keys_exists(element, *keys):
+    '''
+    Check if *keys (nested) exists in `element` (dict).
+    '''
+    if not isinstance(element, dict):
+        raise AttributeError('keys_exists() expects dict as first argument.')
+    if len(keys) == 0:
+        raise AttributeError('keys_exists() expects at least two arguments, one given.')
+
+    _element = element
+    for key in keys:
+        try:
+            _element = _element[key]
+        except KeyError:
+            return False
+    return True
+
+
+class PerfSimpleQueryAnalyzer(BaseResultsAnalyzer):
+
+    collect_last_results_count = 10
+    collect_last_scylla_date_count = 10
+
+    def __init__(self, es_index, es_doc_type):
+        super().__init__(es_index, es_doc_type, query_limit=1000, logger=None)
+
+    def _test_stats(self, test_doc):
+        # check if stats exists
+        if 'results' not in test_doc['_source']:
+            self.log.error('Cannot find the field: results for test id: {}!'.format(test_doc['_id']))
+            return None
+        return test_doc['_source']['results']['perf_simple_query_result'], test_doc["_source"]["versions"][
+            "scylla-server"], test_doc["_source"]["test_details"]
+
+    @staticmethod
+    def get_sorted_results_as_list(results):
+        keys = list(results.keys())
+        keys.sort(reverse=True)
+        return [results[key] for key in keys]
+
+    def check_regression(self, test_id, mad_deviation_limit=0.02, regression_limit=0.05):  # pylint: disable=too-many-locals,too-many-statements
+        doc = self.get_test_by_id(test_id)
+        if not doc:
+            self.log.error('Cannot find test by id: {}!'.format(test_id))
+            return False
+
+        test_stats, versions, test_details = self._test_stats(doc)
+        if not test_stats or not versions:
+            self.log.debug("Could not find test statistics, regression check is skipped")
+            return False
+
+        diff = test_stats['stats']['mad tps'] / test_stats['stats']['median tps']
+        is_deviation_within_limits = bool(diff < mad_deviation_limit)
+        deviation_diff = round((diff) * 100, 2)
+
+        es_base_path = 'hits.hits'
+        es_source_path = es_base_path + '._source'
+        filter_path = ['.'.join([es_base_path, '_id']),
```

Comment:
in future it will be good to have these variables in base class as properties and update filter paths in all inherited classes